Sources/Plasma/CoreLib/hsThread_Unix.cpp

```diff
@@ -75,6 +75,23 @@ int clock_gettime(int clocktype, struct timespec* ts)
 
 /////////////////////////////////////////////////////////////////////////////
 
+void hsThread::SetThisThreadName(const ST::string& name)
+{
+#if defined(HS_BUILD_FOR_APPLE)
+    // The Apple version doesn't take a thread argument and always operates on the current thread.
+    int res = pthread_setname_np(name.c_str());
+    hsAssert(res == 0, "Failed to set thread name");
+#elif defined(HS_BUILD_FOR_LINUX)
+    // On Linux, thread names must fit into 16 bytes, including the terminator.
+    char buf[16];
+    strncpy(buf, name.c_str(), sizeof(buf));
+    buf[sizeof(buf) - 1] = '\0';
+    int res = pthread_setname_np(pthread_self(), buf);
+    hsAssert(res == 0, "Failed to set thread name");
```

Comment:
could a compiler assert by thrown if the name is longer then the limited number of charters?

Reply:
Good question, I'm not really sure. But in the general case, we don't want that to be a hard error - e. g. code that's specific to Windows or Apple can use longer thread names without issues.

Reply:
well i was thinking of some kind of static_assert that possibly called on a macro imbeded in the headers if such macro exists. 

also the static assert would pretty much be ifdefed to linux

Reply:
After doing some research it seems like the limit is imposed by the linux kernel itself and requires a complete recompilation to change.